### PR TITLE
feat(KFLUXBUGS-1417): use stage creds for staged index iib requests

### PR DESCRIPTION
**Pull request by mmalina**
- Setting `.fbc.iibServiceAccountSecret` is no longer allowed (it was never used anyway). Instead, it's based on the stagedIndex setting now.
- If `.fbc.stagedIndex` is `true`, we'll use `iib-service-account-stage`. Otherwise we'll use `iib-service-account-prod`.

The reason is that IIB has a priority queue for requests made with our prod kerberos principal and we were asked to use it only for prod index requests.

Note that this depends on https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/115673

### CODE REVIEW

**Review by theflockers — APPROVED** [review]
lgtm.

**Review by johnbieren — APPROVED** [review]
LGTM, but please don't merge until the MR it depends on is merged (not that e2e is passing anyways)

**Comment by mmalina** [comment]
> LGTM, but please don't merge until the MR it depends on is merged (not that e2e is passing anyways)

Yeah, luckily the app-interface MR is merged now, so this is ready to be merged (if e2e just passed...).

**Pull request merged**
